polyglot-connector-java/src/main/java/com/flipkart/polyglot/jdbc/PolyglotDriver.java

```diff
@@ -0,0 +1,118 @@
+import java.util.Properties;
+import java.util.logging.Logger;
+
+/**
+ * Created by naveen.nahata on 22/03/16.
+ */
+public class PolyglotDriver implements Driver {
+
+    static {
+        try {
+            DriverManager.registerDriver(new PolyglotDriver());
+            DriverManager.registerDriver(new VitessDriver());
+            DriverManager.registerDriver(new PhoenixDriver());
+        } catch (SQLException e) {
+            throw new RuntimeException(
+                    Constants.SQLExceptionMessages.INIT_FAILED + " : " + e.getErrorCode() + " - " + e
+                            .getMessage());
+        }
+    }
+
+    /**
+     * Connect with polyglot driver using url and info properties
+     * @param url
+     * @param info
+     * @return
+     * @throws SQLException
+     */
+    public Connection connect(String url, Properties info) throws SQLException {
+        if(acceptsURL(url)){
+            PolyglotJDBCUrl polyglotJDBCUrl = new PolyglotJDBCUrl(url, info);
+            DataStoreType dataStoreType = polyglotJDBCUrl.getDataStoreType();
+            switch (dataStoreType){
+                case TXNL :
+                    return DriverManager.getConnection(polyglotJDBCUrl.getTxnStroeUrl(), info);
+                case ARCHIVAL:
+                    return DriverManager.getConnection(polyglotJDBCUrl.getArchivalStoreUrl(),info);
+            }
+        }
+        return  null;
+    }
+
+    /**
+     * Return true if url start with jdbc:polyglot
+     * @param url
+     * @return
+     * @throws SQLException
+     */
+    public boolean acceptsURL(String url) throws SQLException {
+        return null != url && url.startsWith(Constants.URL_PREFIX);
+    }
+
+    /**
+     * Get Driver property infos
+     * @param url
+     * @param info
+     * @return
+     * @throws SQLException
+     */
+    public DriverPropertyInfo[] getPropertyInfo(String url, Properties info) throws SQLException {
+        PolyglotJDBCUrl polyglotJDBCUrl = new PolyglotJDBCUrl(url, info);
+        VitessDriver vitessDriver = new VitessDriver();
+        DriverPropertyInfo[] vitessDpi = vitessDriver.getPropertyInfo(polyglotJDBCUrl.getTxnStroeUrl(),info);
+        PhoenixDriver phoenixDriver = new PhoenixDriver();
+        DriverPropertyInfo[] phoenixDpi = phoenixDriver.getPropertyInfo(polyglotJDBCUrl.getArchivalStoreUrl(),info);
+        DriverPropertyInfo[] dpi = (DriverPropertyInfo[]) ArrayUtils.addAll(vitessDpi,phoenixDpi);
+        return dpi;
+    }
+
+    /**
+     * Driver Major version
+     * @return
+     */
+    public int getMajorVersion() {
+        return Constants.DRIVER_MAJOR_VERSION;
+    }
+
+    /**
+     * Driver Minor version
+     * @return
+     */
+    public int getMinorVersion() {
+        return Constants.DRIVER_MINOR_VERSION;
+    }
+
+    public boolean jdbcCompliant() {
+        return Constants.JDBC_COMPLIANT;
+    }
+
+    public Logger getParentLogger() throws SQLFeatureNotSupportedException {
+        return null;
```

Comment:
Shouldn;t this throw featureNotSupportedException. Please refere https://community.oracle.com/thread/2335109?start=0&tstart=0